File: server/Models/VaultKeep.cs
namespace keepr.Models;

public class VaultKeep : DbItem<int>
{
  public int KeepId { get; set; }
  public int VaultId { get; set; }
  public string CreatorId { get; set; }
}


// Creates a Keep with VaultKeepId on it
public class SavedKeep : Keep
{
  public int VaultKeepId { get; set; }
}

File: client/src/models/VaultKeep.js
import { DbItem } from "./DbItem.js";

export class VaultKeep extends DbItem{
  constructor(data){
    super(data);
    this.keepId = data.keepId;
    this.vaultId = data.vaultId;
    this.creatorId = data.creatorId;
  }
}